### PR TITLE
Allow completing failed deposit/close

### PR DESCRIPTION
**Pull request by rmgaray**
Depends on #145

### CODE REVIEW

File: frontend/index.d.ts [hunk 1]
```diff
@@ -7,8 +7,10 @@ export declare class Pool<T> {
 
   constructor(config: SdkConfig, args: T, address: string);
 
-  deposit(amount: BigInteger, batchSize: BigInteger, idxArray: int[]): Promise<int[]>;
+  deposit(amount: BigInteger, batchSize: BigInteger): Promise<IncompleteDeposit | null>;
+  completeDeposit(incompleteDeposit: IncompleteDeposit, batchSize: BigInteger): Promise<IncompleteDeposit | null>;
```
Comment (nazrhom) [inline frontend/index.d.ts:11]: shouldn't this take `mbIncompleteDeposit = IncompleteDeposit | null`? Anything that can be returned by `deposit` should be a valid argument for `completeDeposit`.
I guess the client code will want to do something like:

```
mbCompleteDeposit = await deposit(...)
while mbCompleteDeposit != null {
  mbCompleteDeposit = await completeDeposit(mbCompleteDeposit,..)
```

**Comment by nazrhom** [comment]
@rmgaray Implementation looks good, only got a slight question around the external API, but other than that should be good to go!

File: frontend/src/SdkApi.purs [hunk 2]
```diff
@@ -728,18 +734,18 @@ callGetNodeTime cfg = fromAff
 
 -- We export constructors and consumers of `Maybe` for the JS code
 
-callJust :: forall a. a -> Maybe a
-callJust = Just
+callJust :: forall a. a -> Effect (Promise (Maybe a))
+callJust = Promise.fromAff <<< pure <<< Just
```
Comment (rmgaray) [inline frontend/src/SdkApi.purs:738]: Why do we need to make this function impure? (as well as `callNothing` and `callConsumeMaybe`).

**Review by rmgaray — COMMENTED** [review]
LGTM :+1: . I have only one question about the changed signature of `callNothing` and friends.

File: frontend/src/SdkApi.purs (hunk 2 above)
Comment (nazrhom) [reply frontend/src/SdkApi.purs]: @rmgaray you are right, had some issues initially and thought you need to make it impure when going from JS to PS... but that is not the case

**Review by nazrhom — APPROVED** [review]

**Pull request merged**